kvmagent/kvmagent/plugins/vipqos.py

```
@@ -192,7 +192,14 @@ def _find_classid_with_port(self, nic, ns_cmd, port):
     @in_bash
     def _delete_vipqos_rule_from_nic(self, nic, ns_cmd, port):
         if port == 0:
-            self._change_vipqos_default_rule_bandwidth(nic, ns_cmd, self.VIPQOS_DEFAULT_RULE_BANDWIDTH)
+            # Check if there are per-port class rules (classid > 1:1)
+            remaining = bash_o("{{ns_cmd}} tc class show dev {{nic}} | grep -v 'classid 1:1 ' | grep 'classid 1:'")
```

Ruff (0.15.0)

[error] `bash_o` may be undefined, or defined from star imports

(F405)